File: leetcode/1015. Smallest Integer Divisible by K/3.py
# Complexity (k = number to find a divident)
# Time complexity: O(k)
# Space complexity: O(1)

class Solution:
    def smallestRepunitDivByK(self, K: int) -> int:
        n = 0
        for n_len in range(1, K+1):
            n = (n * 10 + 1) % K
            if n == 0:
                return n_len
        return -1

File: leetcode/215. Kth Largest Element in an Array/3.cpp
/**
 * Complexity (n = input array size, k = number of closest points to pick)
 * Time complexity: O(k * log(n))
 * Space complexity: O(n)
 */

#include <bits/stdc++.h>
using namespace std;

class Solution {
   public:
    int findKthLargest(vector<int>& nums, int k) {
        std::nth_element(nums.begin(), nums.begin() + k - 1, nums.end(), std::greater<int>());
        return *(nums.begin() + k - 1);
    }
};
